### PR TITLE
wip: Allow UDF Decorators to create AbstractUDF

### PR DESCRIPTION
**Pull request by LordDarkula**
- Add `UDFService` class that allows user to create a UDF with decorators and generates a subcless of `AbstractUDF`
- Rename `Array_Count` to `ArrayCount`
- Add pickle file read capability to `path_to_class`

TODO: Add command for decorator UDF registration

### CODE REVIEW

File: eva/udfs/udf_service.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        def wrapper_forward(self, frames: ArrayLike):
+            return func(frames)
+
+        self._forward = wrapper_forward
+        return wrapper_forward
+
+    def __call__(self, *args: Any, **kwds: Any) -> Any:
+        return self._forward(args[0])
+
+    def create_udf(self) -> Type[AbstractUDF]:
```
Comment (gaurav274) [inline eva/udfs/udf_service.py:60]: Are we getting rid of create UDF command?

Comment (LordDarkula) [reply eva/udfs/udf_service.py:60]: `create_udf` outputs an in-memory UDF class that can be instantiated and used. Right now it is the only way of interacting with a UDF created with decorators. Maybe in the future when I figure out registration this method will become unnecessary.

Comment (gaurav274) [reply eva/udfs/udf_service.py:60]: Wait, I'm confused. Are we replacing the current CREATE UDF command with some new command?

File: test/integration_tests/test_udf_decorator_executor.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    CatalogManager().reset()
+    create_sample_video(NUM_FRAMES)
+    load_query = """LOAD FILE 'dummy.avi' INTO MyVideo;"""
+    execute_query_fetch_all(load_query)
+    yield
+    file_remove("dummy.avi")
+
+
+@pytest.fixture(scope="session")
+def dummy_object_detector():
```
Comment (gaurav274) [inline test/integration_tests/test_udf_decorator_executor.py:40]: What is going on in this test case?

File: eva/udfs/udf_service.py (hunk 1 above)
Comment (LordDarkula) [reply eva/udfs/udf_service.py]: Does the updated name and docstring clear up confusion? This method has nothing to do with the `CREATE UDF` SQL command.

File: eva/udfs/udf_service.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import functools
+from enum import Enum, auto
+from typing import Any, Callable, List, Optional, Type
+
+from numpy.typing import ArrayLike
+
+from eva.udfs.abstract.abstract_udf import AbstractClassifierUDF, AbstractUDF
+
+
+class FrameType(Enum):
```
Comment (gaurav274) [inline eva/udfs/udf_service.py:25]: Is there a plan to use this Enum?

File: eva/udfs/udf_service.py (hunk 1 above)
Comment (gaurav274) [reply eva/udfs/udf_service.py]: Which docstring are you referring to? 
And in my understanding, `create_udf` should not be exposed to the user. They should only specify forward and setup functions and we should be able to create the UDF from it.

**Comment by jarulraj** [comment]
@LordDarkula @IshSiva Is this PR still needed with the recent changes? (like #607)

**Comment by IshSiva** [comment]
I think this can be closed as we have changed the implementation of decorators.

**Pull request closed without merging**